pyproject.toml

```diff
@@ -35,6 +35,7 @@ termcolor = "^2.4.0"
 spyne = "^2.14.0"
 lxml = "^5.2.2"
 xmltodict = "^0.13.0"
+pytest = "^8.3.3"
```

Comment:
Pytest is already in the dev dependencies so doesn't need to be added to the pyproject.toml. (If you check the contribution guide in the repo it has instructions for installing dev dependencies). Can you remove this line and regenerate the poetry.lock?

Reply:
Also, by installing the dev dependencies you'll get pre-commit and ruff installed. These are really neat tools that will check your code for some common style errors (number of new lines between functions, max line length) and auto-correct them when you run `git commit`.

Reply:
poetry.lock regenerated and dev dependencies installed !